contracts/most/Utmost.sol

```diff
@@ -0,0 +1,157 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../proxies/MasterCopyNonUpgradable.sol";
+import "./GenesisUtmost.sol";
+import "../utilityToken/UtilityToken.sol";
+
+/**
+ * @title Utmost contract implements UtilityToken.
+ *
+ * @dev Utmost functions as the base coin to pay for gas consumption on the
+ *      auxiliary chain.
+ */
+contract Utmost is MasterCopyNonUpgradable, GenesisUtmost, UtilityToken {
+
+    /* Events */
+
+    /** Emitted whenever the ERC20 Utmost is converted to base coin Utmost. */
+    event TokenUnwrapped(
+        address indexed _account,
+        uint256 _amount
+    );
+
+    /** Emitted whenever the base coin Utmost is converted to ERC20 Utmost. */
+    event TokenWrapped(
+        address indexed _account,
+        uint256 _amount
+    );
+
+
+    /** Constants */
+
+    /** Token Symbol */
+    string public constant TOKEN_SYMBOL = "UM";
+
+    /** Token Name */
+    string public constant TOKEN_NAME = "Utmost";
+
+    /** Token Decimal */
+    uint8 public constant TOKEN_DECIMALS = 18;
+
+
+    /* External Functions */
+
+    /**
+     * @notice Sets up the genesisTokenSupply and calls UtilityToken.setup.
+     *
+     * @dev This function must be called only once. This check is done in setup
+     *      function of UtilityToken.
+     *
+     * @param _genesisTokenSupply Initial token supply.
+     * @param _consensusCogateway ConsensusCogateway contract address.
+     *
+     */
+    function setup(
+        uint256 _genesisTokenSupply,
+        address _consensusCogateway
+    )
+        external
+    {
+        UtilityToken.setup(
+            TOKEN_SYMBOL,
+            TOKEN_NAME,
+            TOKEN_DECIMALS,
+            _genesisTokenSupply,
+            _consensusCogateway
+        );
+
+        genesisTotalSupply = _genesisTokenSupply;
+    }
+
+    /**
+     * @notice Unwrap converts ERC20 Utmost to base coin.
+     *
+     * @dev  This contract's base coin balance must always be greater than
+     *       unwrap amount.
+     *
+     * @dev Function requires:
+     *      - Amount should be non-zero
+     *      - Caller Utmost token balance should be greater than amount
+     *
+     * @param _amount Amount of ERC20 Utmost token to convert to base coin.
+     */
+    function unwrap(
+        uint256 _amount
+    )
+        external
+        returns (bool success_)
```

Comment:
So we need a return value? Where it will be used?

Reply:
This is consistent with our token contract methods implementations. On successful execution of wrap and unwrap, true is returned. 
**UtilityToken**: https://github.com/mosaicdao/mosaic-1/blob/develop/contracts/utilityToken/UtilityToken.sol#L115
 **ERC20Token**: https://github.com/mosaicdao/mosaic-1/blob/develop/contracts/utilityToken/ERC20Token.sol#L133
I am not sure if we should remove this consistency.

This is also in align with OSTPrime implementation in Mosaic-0 where we agreed to return true on wrap and unwrap
https://github.com/mosaicdao/mosaic-contracts/pull/533#discussion_r240216772

Do you see any concerns with returning `success_` = true?

Reply:
My question here was because I don't see the code where it can be `false`. 
Either it `reverts` or returns `true`.

Reply:
I remember we had discussion previously of an `EIP` which we followed for return. Not able to find it now. Will update here if i find it. Removed the return for now.